### PR TITLE
DDF-2888 add transformers to templated docs

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

- adds template for transformers and related components.

#### Who is reviewing it?

@mcalcote @vinamartin @Lambeaux @jrnorth 

#### Select at least one member from relevant component team(s) from below.

[Docs](https://github.com/orgs/codice/teams/docs)

#### Choose 2 committers to review/merge the PR.

@clockard
@coyotesqrl
@lessarderic
@pklinef
@shaundmorris

#### How should this be tested?
verify all sections are showing up in `jdocumentation.pdf` (attached)

#### Any background context you want to provide?

The `*-contents.adoc` files are directly copied from the existing modules. Headers have been added and titles reformatted. Content review _very_ optional.

#### What are the relevant tickets?
[DDF-2888](https://codice.atlassian.net/browse/DDF-2888)

#### Screenshots (if appropriate)

[jdocumentation.pdf](https://github.com/codice/ddf/files/972059/jdocumentation.pdf)

#### Checklist:
- [x] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: distribution/docs/src/main/jdocs/content/_transformers/atom-query-response-transformer-contents.adoc [hunk 1]
```diff
@@ -0,0 +1,175 @@
+:title: Atom Query Response Transformer
+:type: transformer
+:status: published
+:link: _atom_query_response_transformer
+:summary: Transforms a query response into an http://tools.ietf.org/html/rfc4287[Atom 1.0] feed.
+:input:
+:metacard:
+:queryresponse: x
```
Comment (clockard) [inline distribution/docs/src/main/jdocs/content/_transformers/atom-query-response-transformer-contents.adoc]: is `queryresponse: x` right?
Could we make this `sub-type` and have values of input, metacard or queryresponse?

File: distribution/docs/src/main/jdocs/content/_transformers/mime-type-mapper-intro-contents.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
+:title: Mime Type Mapper Intro
+:type: transformerIntro
+:status: published
+:link: _mime_type_mapper_intro
+:summary: Resolves file extensions to mime types and vice versa.
+:input:
+:metacard:
```
Comment (clockard) [inline distribution/docs/src/main/jdocs/content/_transformers/mime-type-mapper-intro-contents.adoc]: do we need these fields here if none of them are marked?

File: pom.xml [hunk 3]
```diff
@@ -778,7 +778,6 @@
                     <plugin>
                         <groupId>org.owasp</groupId>
                         <artifactId>dependency-check-maven</artifactId>
-                        <version>1.2.9</version>
```
Comment (clockard) [inline pom.xml]: This change seems unrelated to the other changes. does it belong?

Comment (ricklarsen) [inline pom.xml]: it was previously added in error.

File: distribution/docs/src/main/jdocs/content/_transformers/atom-query-response-transformer-contents.adoc (hunk 1 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_transformers/atom-query-response-transformer-contents.adoc]: Just making three separate types out the transformers because it's impossible to be more than one type of transformer.

File: distribution/docs/src/main/jdocs/content/_transformers/mime-type-mapper-intro-contents.adoc (hunk 2 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_transformers/mime-type-mapper-intro-contents.adoc]: It does cause problems if the property doesn't exist, even if it's not set. 

That said, I'm going to simplify this by defining the types differently, so this won't be necessary.

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-query-response-transformers-contents.adoc [hunk 4]
```diff
@@ -0,0 +1,55 @@
+:title: Developing Query Response Transformers
+:type: developingComponent
+:status: published
+:link: _developing_query_response_transformers
+:summary: Creating a custom Query Response Transformer.
```
Comment (vinamartin) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-query-response-transformers-contents.adoc]: Be consistent with the period usage in the summaries.
I'm neither for or against them.

File: distribution/docs/src/main/jdocs/content/_transformers/mime-type-resolver-intro-contents.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
+:title: Mime Type Resolver Intro
+:type: transformerIntro
+:status: published
+:link: _mime_type_resolver_intro
+:summary:
```
Comment (vinamartin) [inline distribution/docs/src/main/jdocs/content/_transformers/mime-type-resolver-intro-contents.adoc]: Summary is blank here, should it be?

File: distribution/docs/src/main/jdocs/content/_developingComponents/developing-query-response-transformers-contents.adoc (hunk 4 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/developing-query-response-transformers-contents.adoc]: I'm for them

File: distribution/docs/src/main/jdocs/content/_transformers/mime-type-resolver-intro-contents.adoc (hunk 5 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_transformers/mime-type-resolver-intro-contents.adoc]: No, It's not getting used by the template yet, but I added a summary anyway.

File: distribution/docs/src/main/jdocs/content/_transformers/pptx-input-transformer-contents.adoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
+:title: PPTX Input Transformer
+:type: transformer
+:subtype: input
+:status: published
+:link: _pptx_input_transformer
+:summary: Translates Microsoft PowerPoint (OOXML only) documents into a Catalog Metacard.
```
Comment (coyotesqrl) [inline distribution/docs/src/main/jdocs/content/_transformers/pptx-input-transformer-contents.adoc]: Should be singular PP document -> Catalog Metacard for consistency.

Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_transformers/pptx-input-transformer-contents.adoc]: I went the other way and made it plural...

**Comment by ricklarsen** [comment]
build now

**Comment by ricklarsen** [comment]
build now

**Comment by ricklarsen** [comment]
build now
